pdns/dnsdistdist/libssl.cc

```diff
@@ -685,6 +697,11 @@ std::unique_ptr<SSL_CTX, void(*)(SSL_CTX*)> libssl_init_server_context(const TLS
     SSL_CTX_sess_set_cache_size(ctx.get(), config.d_maxStoredSessions);
   }
 
+  /* we need to set this callback to acknowledge the server name sent by the client,
+     otherwise it will not stored in the session and will not be accessible when the
+     session is resumed, causing SSL_get_servername to return nullptr */
+  SSL_CTX_set_tlsext_servername_callback(ctx.get(), &libssl_server_name_callback);
```

Comment:
extremely pedantic nit: redundant `&`

Reply:
I personally prefer it that way to make it more clear that we pass the address of the function, and not the result of a call to the function.

Reply:
To my eyes it's clear it isn't a call since no arguments are passed. But fair enough.